java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java

```diff
@@ -754,26 +659,25 @@ else if (this.ksdata.isRhel3() || this.ksdata.isRhel4()) {
             up2datehost = this.session.getSystemRhnHost();
         }
 
-        log.debug("adding perl -npe for /etc/sysconfig/rhn/up2date");
-        if (this.ksdata.isRhel2()) {
+        if (!ConfigDefaults.get().getUserSelectedSaltInstallTypeLabels().contains(ksdata.getInstallType().getLabel())) {
+            log.debug("adding perl -npe for /etc/sysconfig/rhn/up2date");
+            // both rhel 2 and rhel3/4 need the following
             retval.append("perl -npe " +
-                    "'s|^(\\s*(noSSLS\\|s)erverURL\\s*=\\s*[^:]+://)[^/]*/|${1}" +
-                     up2datehost +
-                     "/|' -i /etc/sysconfig/rhn/rhn_register" + NEWLINE);
+                    "'s|^(\\s*(noSSLS\\|s)erverURL\\s*=\\s*[^:]+://)[^/]*/|\\${1}" +
+                    up2datehost +
+                    "/|' -i /etc/sysconfig/rhn/up2date" + NEWLINE);
+
+            if (this.ksdata.getVerboseUp2date()) {
+                retval.append("[ -r /etc/sysconfig/rhn/up2date ] && " +
+                        "sed 's/debug=0/debug=1/' -i /etc/sysconfig/rhn/up2date" +
+                        NEWLINE);
+            }
```

Comment:
This is still calling `perl` which is however now removed (or at least not explicitly added) on line 591.

Is this even needed when according to comment this is for rhel2 to 4?

Reply:
I also do not understand what the condition is checking. :D

Reply:
This if is for traditional client kickstarts on rhel 6 and 7... and those still have perl: I didn't remove it from that code path.

The verbose up2date is just an option the user can click in the UI